pywren/pywren_ibm_cloud/storage/cleaner.py

```diff
@@ -47,6 +47,7 @@ def clean_os_bucket(bucket, prefix, internal_storage):
     while objects_to_delete:
         logger.debug('{} objects found'.format(len(objects_to_delete)))
         total_objects = total_objects + len(objects_to_delete)
-        internal_storage.delete_temporal_data(objects_to_delete)
+        for execute_deletion in internal_storage.delete_temporal_data(objects_to_delete):
+            deletion_result = execute_deletion
         objects_to_delete = internal_storage.list_temporal_data(prefix)
```

Comment:
@omerb01 why we need to replace a delete call with for statement?

Reply:
the delete method in cos.py returns a generator, so we need to iterate over it and execute every 1000 objects deletion

Reply:
@omerb01 I prefer internal cos related logic to be hidden. There should be delete method that delete stuff inside, not return generator. Can you please move the loop logic inside cos implementation and keep simply 
internal_storage.delete_temporal_data(objects_to_delete)
?
This will be cleaner and not related COS. Notice that internal_storage may point to other implementations, not just COS. So generator not always makes sense

Reply:
@omerb01 so let's hide generator inside. It's wrong to return generator for delete method

Reply:
ok, changed the relevant parts